### PR TITLE
test HTTAgent class methods

### PR DESCRIPTION
**Pull request by Andati**
`post`, `urlCanBeAccessWithGivenCredentials` & `fetchWithCredentials` methods

### CODE REVIEW

File: opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java [hunk 1]
```diff
@@ -130,6 +130,7 @@ private HttpURLConnection initializeHttp(String requestURLPath, boolean useBasic
         try {
             urlConnection = initializeHttp(postURLPath, true);
 
+            urlConnection.setDoOutput(true);
```
Comment (githengi) [inline opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java]: what is the reason for this change. 

This flag may not be required for GET requests

Comment (Andati) [reply opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java]: This is in a POST request. A test was failing due to this missing

Comment (githengi) [reply opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java]: I see i thought this was on `initializeHttp` method
